extensions/apps/extensions/src/components/app-routes/index.tsx

```diff
@@ -188,11 +203,34 @@ const devInfoRoute = createRoute({
 const collaboratorsInfoRoute = createRoute({
   getParentRoute: () => infoRootRoute,
   path: '/collaborators',
+  loader: ({ context, params }) => {
+    const { appId } = params;
+    if (!appId) {
+      throw new Error('appId is required');
+    }
+    return {
+      extensionById: defer(getExtensionById(context.decodeAppName(appId))),
+    };
+  },
   component: () => {
     const { appId } = infoRootRoute.useParams();
+    const { extensionById } = collaboratorsInfoRoute.useLoaderData();
     return (
       <CatchBoundary getResetKey={() => 'collaborators_reset'} errorComponent={RouteErrorComponent}>
-        <CollaboratorsPage appId={appId} />
+        <Suspense>
+          <Await promise={extensionById}>
+            {data => (
+              <CollaboratorsPage
+                extensionLogo={selectExtensionLogo(data)}
+                extensionName={selectExtensionName(data)}
+                extensionDisplayName={selectExtensionDisplayName(data)}
+                collaborators={selectExtensionCollaborators(data)}
```

Comment:
on the app model only the dids for collaborators are saved, so you need to fetch the profile data before passing it here, I created a new hook in my PR called use-profiles-list that can be sued for this

Reply:
check this out: https://feat-extension-page-sub-1.akasha-world-framework.pages.dev/@akashaorg/app-extensions/info/complexity-app

Reply:
ok, I see that one of the profile has also name, but I don't understand where it's coming from by looking at the code, it doesn't look like you are fetching profile data anywhere,  and on the model it is not stored. I will take a more in depth look after I fix my PR

Reply:
black magix